File: 3461.check-if-digits-are-equal-in-string-after-operations-i/solution.go
func hasSameDigits(s string) bool {
    b := make([]byte, len(s))
    for i := range s {
        b[i] = s[i] - '0'
    }
    for len(b) > 2 {
        var c []byte
        for i := 1; i < len(b); i++ {
            c = append(c, (b[i - 1] + b[i]) % 10)
        }
        b = c
    }
    return b[0] == b[1]
}

File: 3461.check-if-digits-are-equal-in-string-after-operations-i/solution.kt
class Solution {
    fun hasSameDigits(s: String): Boolean {
        var nums = s.map { it - '0' }
        while (nums.size > 2) {
            nums = nums.zipWithNext { a, b -> (a + b) % 10 }
        }
        return nums[0] == nums[1]
    }
}